### PR TITLE
PEAR-1961: Survival Plot, tooltip and download, display of time

### PR DESCRIPTION
**Pull request by paribartandhakal**
## Description
1. Refactored the code. Created separate files for legibility.
2. Updated tooltip and tsv.

## Checklist
- [N/A] Added proper unit tests
- [N/A] Left proper TODO messages for any remaining tasks
- [N/A] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

### CODE REVIEW

File: packages/portal-proto/src/features/charts/SurvivalPlot/SurvivalPlot.tsx [hunk 1]
```diff
@@ -457,9 +156,15 @@ const ExternalDownloadStateSurvivalPlot: React.FC<SurvivalPlotProps> = ({
       .map((element, index) =>
         element.donors
           .map((row) => {
+            const timeDays = Math.round(row.time * 365.25);
+            const timeMonths = Math.round(row.time * 12);
+            const timeYears = Number(row.time.toFixed(1));
+
```
Comment (paribartandhakal) [inline packages/portal-proto/src/features/charts/SurvivalPlot/SurvivalPlot.tsx]: actual changes in this function.

File: packages/portal-proto/src/features/charts/SurvivalPlot/useSurvival.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+              : `<tspan font-style="italic">S</tspan><tspan font-size="0.7em" baseline-shift="-25%">${
+                  curves.indexOf(curve) + 1
+                }</tspan>`,
+          onMouseEnterDonor: (
+            e,
+            { censored, project_id, submitter_id, survivalEstimate, time = 0 },
+          ) => {
+            const years = Number(time.toFixed(1));
+            const months = Math.round(time * 12);
+            const timeString = `${years} years (${months} months)`;
```
Comment (paribartandhakal) [inline packages/portal-proto/src/features/charts/SurvivalPlot/useSurvival.tsx]: actual changes in this function.

**Comment by wteouchicago** [comment]
## Issues
1. The conversion to months seems off. I have a value of 3478 days (in the TSV). It says "114" months in the tooltip and the TSV. but 114 / 30 should be 116 (rounded to int). 
- don't use the converted-to-years value then multiply by 12; convert from the original days value or we'll see discrepancies.

2. Ultra minor. Don't bother fixing unless it's super trivial.
- when the tooltip has exactly 1 month or exactly 1 year, can we use the singular of "year" and "month"?
Screenshot 2024-07-18 at 4 10 41 PM
Screenshot 2024-07-18 at 4 10 27 PM

**Comment by wteouchicago** [comment]
looks good! 👍

**Review by amlehman — APPROVED** [review]

File: packages/portal-proto/src/features/charts/SurvivalPlot/types.ts
Comment (craigrbarnes) [inline packages/portal-proto/src/features/charts/SurvivalPlot/types.ts:1]: Nice refactoring of the Survival plot, it has been needed for a while 😄

**Review by craigrbarnes — APPROVED** [review]
Looks fine to me.

**Pull request merged**